### DISCUSSION TITLE
Exporting Filtered Data from Buefy Tables in Vue

### DISCUSSION THREAD

**Discussion by ishack79** [buefy/buefy#3909] [General]
Hi all,

I'm working on a Vue project with tables implemented using Buefy. These tables have searchable columns and my Vue component has a feature to export the displayed data to an Excel file. However, I'm encountering an issue: the exported data doesn't reflect the filtered state of the table. The data, stored in an array within my Vue component, remains unaltered even when I apply filters to the columns.

Can anyone guide me on how to export the data as it appears post-filtering in the Buefy table?

Thanks, Isaac.

**Comment by ishack79** [comment]
this.$refs.table.newData works.

**Comment by agb2008** [comment]
I was able to export filtered Buefy 3.0.3 BTable data using:

```<script setup>

import { ref } from 'vue'

const myTable = ref(null)
// rest of code
</script>
```
```
<template>
     <b-table ref="myTable"
             :data="table"
             
      </b-table>
</template>
```

In that case 
number of filtered data items could displayed using: **myTable.newDataTotal**
and array with filtered data: **myTable.newData**

_vue: 3.5.22, buefy: 3.0.3, SaveToXlsx: custom version_
